PAbishai/Ride-My-Way
# Refactored and Restructured Code Base 159356407

**What does this PR do**
Restructure and Refactor the code base for easy maintenance and scalability

**Description of tasks to be completed?**
Separate the tests based on the endpoints they are testing
The end result was three files:
 - test_auth
 - test_rides
 - test_requests

Separate the models and resources based on the classes they instantiated from
The end result was
models:

- user_model
- ride_model
- requests_model

resources

- auth_resources
- ride_resources
- requests_resources

**How should this be manually tested?**
Clone the repo 
Open the app folder to view the models and resources
Open the tests folder to view the tests
To test the endpoints:
Install postgress
Create a database ride-my-way
`createdb ride-my-way`
Create a virtualenv 
`#virtualenv Ride-My-Way`
cd into the repo and activate it
run
`pip install -r requirements.txt`
cd into the resources folder
edit 'models.py'
add your postgres username and password to conn_string
run 
`FLASK_APP=run.py FLASK_DEBUG=1 flask run`
localhost:5000[](localhost:5000) should display the api documentation
